### ISSUE TITLE
Add Web Push Notifications support

### ISSUE THREAD

**Issue by contrapunctus-1** [conversejs/converse.js#1519]
Contacts who use Converse.JS as their only client get notified of messages only when they open Converse.JS and log in. To be able to reach them regardless of that, the current solution is for them to use a mobile client.

If Web Push Notifications can help improve this, please add support for them in Converse.JS.

**Comment by laszlovl** [comment]
During a recent sprint, we managed to create a working POC for webpush in Converse. If you're interested in the (extremely rough) code you can check https://github.com/laszlovl/converse.js/commits/push and https://modules.prosody.im/mod_web_push.html.

**Comment by Echolon** [comment]
@jcbrand thats not fully implemented yet right?

**Comment by jcbrand** [comment]
No, the existing work is here: https://github.com/conversejs/converse.js/pull/2153

**Comment by jcbrand** [comment]
I'm not sure the `in progress` label is appropriate here, since no-one is actually working on this currently.

**Comment by jcbrand** [comment]
For reference:
https://www.process-one.net/blog/webpush-support-on-your-fluux-io-instance/
